### PR TITLE
[TextFields] New text area style.

### PR DESCRIPTION
**Pull request by willlarche**
Dependent on #1816 , #1812 , #1814 , #1813 .

### CODE REVIEW

File: components/TextFields/src/MDCTextInputControllerTextArea.m [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+  // The amount of space underneath the underline depends on whether there is content in the
+  // underline labels.
+  CGFloat underlineLabelsOffset = 0;
+  if (self.textInput.leadingUnderlineLabel.text.length) {
+    underlineLabelsOffset =
+    MDCCeil(self.textInput.leadingUnderlineLabel.font.lineHeight * 2.f) / 2.f;
+  }
+  if (self.textInput.trailingUnderlineLabel.text.length || self.characterCountMax) {
+    underlineLabelsOffset =
+    MAX(underlineLabelsOffset,
```
Comment (ajsecord) [inline components/TextFields/src/MDCTextInputControllerTextArea.m]: +4

File: components/TextFields/src/MDCTextInputControllerTextArea.m [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+  self.textInput.underline.alpha = 0;
+
+  if (!self.placeholderTop) {
+    self.placeholderTop = [NSLayoutConstraint constraintWithItem:self.textInput.placeholderLabel attribute:NSLayoutAttributeTop relatedBy:NSLayoutRelationEqual toItem:self.textInput attribute:NSLayoutAttributeTop multiplier:1 constant:MDCTextInputTextFieldTextAreaFullPadding];
+    self.placeholderTop.priority = UILayoutPriorityDefaultHigh;
+    self.placeholderTop.active = YES;
+  }
+}
+
+// Measurement from bottom to underline center Y
```
Comment (ajsecord) [inline components/TextFields/src/MDCTextInputControllerTextArea.m]: Period at end.

File: components/TextFields/src/MDCTextInputControllerTextArea.m [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+  self.textInput.borderView.borderPath.lineWidth = self.textInput.isEditing ? 2 : 1;
+}
+
+- (void)updateLayout {
+  [super updateLayout];
+
+  if (!self.textInput) {
+    return;
+  }
+
```
Comment (ajsecord) [inline components/TextFields/src/MDCTextInputControllerTextArea.m]: Repeat NSAssert() and release-mode guard here.

**Review by ajsecord — CHANGES_REQUESTED** [review]

File: components/TextFields/src/MDCTextInputControllerTextArea.m (hunk 3 above)
Comment (willlarche) [reply components/TextFields/src/MDCTextInputControllerTextArea.m]: Done and improved use of the type below.

File: components/TextFields/src/MDCTextInputControllerTextArea.m (hunk 2 above)
Comment (willlarche) [reply components/TextFields/src/MDCTextInputControllerTextArea.m]: Done!

File: components/TextFields/src/MDCTextInputControllerTextArea.m (hunk 1 above)
Comment (willlarche) [reply components/TextFields/src/MDCTextInputControllerTextArea.m]: Same improvement as previous PR.

**Review by ajsecord — APPROVED** [review]

**Pull request merged**